src/iqm/cirq_iqm/devices/iqm_device_metadata.py

```diff
@@ -89,66 +94,55 @@ def resonator_set(self) -> FrozenSet[NamedQid]:
+        qubits = tuple(NamedQid(qb, dimension=2) for qb in architecture.qubits)
         resonators = tuple(
-            NamedQid(qb, dimension=cls.RESONATOR_DIMENSION)
-            for qb in architecture.qubits
-            if not qb.startswith(cls.QUBIT_NAME_PREFIX)
+            NamedQid(cr, dimension=cls.RESONATOR_DIMENSION) for cr in architecture.computational_resonators
         )
-        connectivity = tuple(
-            tuple(
+
+        def get_qid_locus(locus: Iterable[str]) -> tuple[NamedQid, ...]:
+            """Converts locus component names to Qids."""
+            return tuple(
                 (
-                    NamedQid(qb, dimension=2)
-                    if qb.startswith(cls.QUBIT_NAME_PREFIX)
-                    else NamedQid(qb, dimension=cls.RESONATOR_DIMENSION)
+                    NamedQid(component, dimension=2)
+                    if component in qubits
+                    else NamedQid(component, dimension=cls.RESONATOR_DIMENSION)
                 )
-                for qb in edge
+                for component in locus
             )
-            for edge in architecture.qubit_connectivity
+
+        # connectivity consists of all arity-2 gate loci in the DQA
+        connectivity = tuple(
+            get_qid_locus(locus)
+            for gate_info in architecture.gates.values()
+            for locus in gate_info.loci
+            if len(locus) == 2
         )
         operations: dict[type[cirq.Gate], list[tuple[NamedQid, ...]]] = {
-            cirq_op: [
-                tuple(
-                    (
-                        NamedQid(qb, dimension=2)
-                        if qb.startswith(cls.QUBIT_NAME_PREFIX)
-                        else NamedQid(qb, dimension=cls.RESONATOR_DIMENSION)
-                    )
-                    for qb in args
-                )
-                for args in qubits
-            ]
-            for iqm_op, qubits in architecture.operations.items()
-            for cirq_op in _IQM_CIRQ_OP_MAP[iqm_op]
-        }
-        return cls(qubits, connectivity, operations=operations, resonators=resonators)
-
-    def to_architecture(self) -> QuantumArchitectureSpecification:
-        """Returns the architecture specification object created based on device metadata."""
-        qubits = tuple(qb.name for qb in self._qubit_set)
-        resonators = tuple(qb.name for qb in self.resonator_set)
-        connectivity = tuple(tuple(qb.name for qb in edge) for edge in self.nx_graph.edges())
-        operations: dict[str, list[tuple[str, ...]]] = {
-            iqm_op: [tuple(qb.name for qb in args) for args in qubits]
-            for cirq_op, qubits in self.operations.items()
-            for iqm_op, cirq_ops in _IQM_CIRQ_OP_MAP.items()
-            if cirq_op in cirq_ops
+            cirq_op: [get_qid_locus(locus) for locus in gate_info.loci]
+            for gate_name, gate_info in architecture.gates.items()
+            for cirq_op in _IQM_CIRQ_OP_MAP[gate_name]
```

Comment:
`operations` contains the full locus info for all gates, but it's not used for routing as far as I can tell.
`connectivity` is used (in the form of `IQMDevice._metadata.nx_graph`), but it does not represent any single 2q gate type (seems like the router assumes there is just one 2q gate).

Also, `connectivity` here will contain duplicate loci/edges. When it's converted into a graph the duplicates vanish.

Reply:
Right. As discussed, this problem already existed before this PR and is due to Cirq limitations. Probably can't solve it here. We can still look into a solution for it, but probably the PR should not be blocked from being merged because of this.

Reply:
Does this connectivity work for Deneb-style devices? Where the router should not have the MOVE gate edges. If connectivity will turn into a an `nx_graph` does it makes sense to make labeled edges where the labels correspond to gates? Or is that something the router cannot work with either?

Reply:
So the connectivity is used to build the `nx_graph` this is used in `route_circuit` by removing the resonators from it and then using Cirq's routing pass. After that, the MOVE gates are added. So there is no notion of gates in the `route_circuit` but the case of MOVE gates is handled through the resonator.